### ISSUE TITLE
[CS] There is no way of knowing if a correct xr_session has been created

### ISSUE THREAD

**Issue by DirkSonguer** [maluoi/StereoKit#3]
**Issue**: With C# it's not possible to know if the OpenXR initialization has succeeded. The application will just run silently.

Steps to reproduce:
- Don't attach a Windows MR headset
- Run the StereoKitTest application

**Result**: The application starts normally, however nothing happens. As a developer I have no way of knowing if the application has a correct OpenXR xr_session running.

**Expected**: There is either an exception if no xr_session could be created or some other way of knowing.

**Details**: With CPP it is possible to know if the initialization has failed as sk_init() returns a bool. With C# there is no such thing as the constructor is just calling sk_init() and not listening to any return.

**Suggestion**: Change StereoKit class constructor to:
``` csharp
        public StereoKitApp(string name, Runtime runtime)
        {
            if (!sk_init(name, runtime))
            {
                throw new Exception("StereoKit: Could not initialize " + runtime.ToString() + " runtime");
            }
        }
```

**Proposal**: Add exceptions to StereoKitC library to let them properly bubble up, regardless of CPP or C#. At least for the fatal ones (like creating xr_session, events, actions, rendering etc.)

**Comment by maluoi** [comment]
Something is definitely needed here! I remember writing that example and thinking, 'Huh, I can't check errors out of a constructor, can I?'. Then I moved on. There's definitely a lot of error checking I've deferred so far, it would be really nice to have a good logging system too.

I'm not a fan of catching exceptions, it falls into the realm of 'Magic! Where Did That Functionality Come From?' to me. So I generally try to avoid throwing them unless I actually intend to crash an application. I much prefer error codes!

So I think the core trouble here is that the constructor is doing a little too much work! I think my preference would be either a static StereoKitApp.Create() that can return null, or an instance member StereoKitApp.Initialize() that can return an error code. That and better logging :)

**Comment by DirkSonguer** [comment]
Errors out of constructors are Exceptions 😅

Exceptions are only magic if you use them only once or twice and they come as a surprise. If you are using them throughout then they can be expected and people should code for it. As for error codes: This is what exception types are for.

If you want to avoid them at any cost, then I agree that having a separate StereoKitApp.Init() function would better mirror the C behavior:

``` csharp
        public StereoKitApp()
        {
                // This would be empty now? 
        }

        public Init(string name, Runtime runtime)
        {
            return sk_init(name, runtime);
        }
```

Then I would appreciate blocking any other functionality until the initialization is done, so some kind of `public bool isInitialized` for me to check.

What do you think?
